spec.bs

```diff
@@ -281,6 +281,23 @@ To <dfn>look up penultimate redemption</dfn> given an [=/origin=] |issuer| and a
 1. If [=redemptionTimes=][(|issuer|,|topLevel|)] [=map/exists=], let |penultimateRedemption| be the |penultimateRedemption| field of the tuple [=redemptionTimes=][(|issuer|,|topLevel|)].
 1. Return |penultimateRedemption|.
 
+A user agent has <dfn>redemptionRecords</dfn>, a [=map=] where the keys are a [=tuple=] (|issuer|, |topLevel|), and the values are a [=tuple=] (|record|, |expiration|).
+
+To <dfn>record a redemption record</dfn> given an [=/origin=] |issuer|, an [=/origin=] |topLevel|, a [=byte sequence=] |header|, and a [=duration=] |lifetime|, run the following steps: 
+
+1. If |lifetime| is zero, return.
+1. Let |currentTime| be the current date and time.
```

Comment:
```suggestion
1. Let |currentTime| be the current date and time in milliseconds since 01 January, 1970 UTC.
```

given that you're doing arithmetic on it specifying that seems sensible.

(I think we should re-evaluate spec best practices around time overall for this spec but I'm okay with just making things clear enough for potential future implementers for now)